smartsim/entity/model.py

```diff
@@ -111,6 +118,18 @@ def attach_generator_files(self, to_copy=None, to_symlink=None, to_configure=Non
         to_configure = init_default([], to_configure, (list, str))
         self.files = EntityFiles(to_configure, to_copy, to_symlink)
 
+    def params_to_args(self):
+        """Convert parameters to command line arguments and update run settings.
+        """
+        for param in self.params_as_args:
+            if not param in self.params:
+                raise SSConfigError(f"Tried to convert {param} to command line argument " +
+                                    f"for Model {self.name}, but its value was not found in model params")
+            if self.run_settings is None:
```

Comment:
will this case ever happen?

Reply:
@Spartee it seems like we don't check whether `run_settings` is `None`, thus a very bad user might initialize a `Model` with `run_settings=None`. I don't know if there is any reason why we are allowing this, otherwise we might want to guard against it in another part of the code?

Reply:
yea I think step creation would just fail, but when we go to the server arch I think thisll be more important.